inc/saiarsqualitymap.h

```diff
@@ -0,0 +1,409 @@
+/**
+ * Copyright (c) 2014 Microsoft Open Technologies, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *    not use this file except in compliance with the License. You may obtain
+ *    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc., Marvell International Ltd.
+ *
+ * @file    saiarsqualitymap.h
+ *
+ * @brief   This module defines SAI QOS Maps interface
+ */
+
+#if !defined (__SAIARSQUALITYMAP_H_)
+#define __SAIARSQUALITYMAP_H_
+
+#include <saitypes.h>
+
+/**
+ * @defgroup SAIARSQUALITYMAP SAI - ARS quality map specific API definitions.
+ *
+ * @{
+ */
+
+/**
+ * @brief Adaptive Routing and Switching quality measure algorithm
+ */
+typedef enum _sai_ars_quality_map_algo_t
+{
+    /** Exponentially weighted moving average */
+    SAI_ARS_QUALITY_MAP_ALGO_EWMA,
+
+} sai_ars_quality_map_algo_t;
+
+/**
+ * @brief Enum defining attributes for ARS quality map.
+ */
+typedef enum _sai_ars_quality_map_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_ARS_QUALITY_MAP_ATTR_START,
+
+    /**
+     * @brief ARS algorithm used for quality computation
+     *
+     * @type sai_ars_quality_map_algo_t
+     * @flags CREATE_AND_SET
+     * @default SAI_ARS_QUALITY_MAP_ALGO_EWMA
+     */
+    SAI_ARS_QUALITY_MAP_ATTR_ALGO = SAI_ARS_QUALITY_MAP_ATTR_START,
+
+    /**
+     * @brief Sampling interval in microseconds of data for quality measure computation
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 16
+     */
+    SAI_ARS_QUALITY_MAP_ATTR_SAMPLING_INTERVAL,
+
+    /**
+     * @brief Random seed for adaptive routing and switching
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_ARS_QUALITY_MAP_ATTR_ARS_RANDOM_SEED,
+
+    /**
+     * @brief Maximum number of ECMP ARS groups
+     *
+     * @type sai_uint32_t
+     * @flags READ_ONLY
+     */
+    SAI_ARS_QUALITY_MAP_ATTR_ECMP_ARS_MAX_GROUPS,
```

Comment:
In the current version of this proposal, several attributes of ARS_QUALITY_MAP do not logically belong to "Quality Map".

Reply:
Lets discuss in community meeting

Reply:
Change ars quality map object name to ars profile object.

Reply:
Chnaged the  ARS quality map object to ARS profile object.
https://github.com/opencomputeproject/SAI/pull/1658/commits/05e9558fe4d467f3b0881d6e6ca3e57b0bd923ba